Routers/usersRouter.js

```diff
@@ -0,0 +1,115 @@
+//create router
+const express = require('express');
+const router = express.Router();
+
+//grab helper functions
+const usersDb = require('../data/helpers');
+
+//import bcryptjs for hashing
+const bcrypt = require('bcryptjs');
+//import jsonwebtoken 
+const jwt = require('jsonwebtoken');
+
+
+//**TOKEN GENERATOR FUNCTION */
+const secret = 'williewonka';  //placed in global for use in verification middleware
+function tokenGenerator(user){
+    const payload = {
+        username: user.username,
+        department: user.department
+    }
+    
+    const options = {
+        expiresIn: '1h',
+        jwtid: '12345'  //jti
+    }
+    
+    return jwt.sign(payload, secret, options);
+}
+
+//***VERIFICATION OF TOKEN MIDDLEWARE**** */
+function verifyUser(req, res, next){
+    //read token string from authorization header
+    //if user not logged in there will be no token
+    const token = req.headers.authorization;
+
+    if(token){
+        //verify the token
+        jwt.verify(token, secret, (err, decodedToken) =>{
+            if(err){
+                res.status(401).json({error: "Invalid Token"})
+            }else{
+                next();
+            }
+        })
+    } else{
+        res.status(401).json("No Token Provided")
+    }
+
+}
+
```

Comment:
Good job handling the whole JWT life-cycle Christopher 👍 This is really the core of this module, and you've got the steps down.